### PR TITLE
fix some ui problems

### PR DESCRIPTION
**Pull request by geor-kasapidi**
fix autolayout and preview layer orientation. also fix some small things and warnings.

### CODE REVIEW

File: Sources/Utils/Dropdown/DropdownController.swift [hunk 1]
```diff
@@ -11,7 +11,7 @@ class DropdownController: UIViewController {
   lazy var blurView: UIVisualEffectView = self.makeBlurView()
 
   var animating: Bool = false
-  var expanding: Bool = true
+  var expanding: Bool = false
```
Comment (zenangst) [inline Sources/Utils/Dropdown/DropdownController.swift]: Why did you change the default here? Will this have an impact on existing projects?

File: Sources/Utils/Extensions/AVAsset+Extensions.swift [hunk 2]
```diff
@@ -55,15 +55,15 @@ extension AVAsset {
 
   var g_videoDescription: CMFormatDescription? {
     if let object = tracks(withMediaType: AVMediaTypeVideo).first?.formatDescriptions.first {
-      return object as! CMFormatDescription
+      return (object as! CMFormatDescription)
     }
```
Comment (zenangst) [inline Sources/Utils/Extensions/AVAsset+Extensions.swift]: Perhaps we could refactor this method to do the casting in the if statement so that we get rid of the force unwrapping. What do you think?

```swift
guard let object = tracks(withMediaType: AVMediaTypeVideo).first?.formatDescriptions.first as? CMFormatDescription else {
  return nil
}

return object
```

File: Sources/Utils/Extensions/AVAsset+Extensions.swift [hunk 3]
```diff
@@ -55,15 +55,15 @@ extension AVAsset {
...
 
     return nil
   }
 
   var g_audioDescription: CMFormatDescription? {
     if let object = tracks(withMediaType: AVMediaTypeAudio).first?.formatDescriptions.first {
-      return object as! CMFormatDescription
+      return (object as! CMFormatDescription)
     }
```
Comment (zenangst) [inline Sources/Utils/Extensions/AVAsset+Extensions.swift]: I think we can do the same thing here as my previous comment.

File: Sources/Utils/Dropdown/DropdownController.swift (hunk 1 above)
Comment (geor-kasapidi) [reply Sources/Utils/Dropdown/DropdownController.swift]: In the previous commit this flag was set to true, and after the dropdownController.toggle () method was called. Just a little optimization. Current projects are not affected in any way

Comment (zenangst) [reply Sources/Utils/Dropdown/DropdownController.swift]: Nice, just wanted to make sure 🚀

File: Sources/Utils/Extensions/AVAsset+Extensions.swift [hunk 4]
```diff
@@ -54,18 +54,18 @@ extension AVAsset {
...
 
   var g_audioDescription: CMFormatDescription? {
-    if let object = tracks(withMediaType: AVMediaTypeAudio).first?.formatDescriptions.first {
-      return object as! CMFormatDescription
+    guard let object = tracks(withMediaType: AVMediaTypeAudio).first?.formatDescriptions.first else {
+      return nil
     }
 
-    return nil
+    return (object as! CMFormatDescription)
```
Comment (zenangst) [inline Sources/Utils/Extensions/AVAsset+Extensions.swift]: I think you can move the casting into the `guard` as well to get rid of the force unwrapping :)

File: Sources/Utils/Extensions/AVAsset+Extensions.swift (hunk 2 above)
Comment (geor-kasapidi) [reply Sources/Utils/Extensions/AVAsset+Extensions.swift]: I've changed 'if let' to 'guard let', but it's just a syntax sugar - Conditional downcast to CoreFoundation type 'CMFormatDescription' will always succeed :) 'guard' let looks better

File: Sources/Utils/Extensions/AVAsset+Extensions.swift (hunk 4 above)
Comment (geor-kasapidi) [reply Sources/Utils/Extensions/AVAsset+Extensions.swift]: swift compiler gives an error - "Conditional downcast to CoreFoundation type 'CMFormatDescription' will always succeed". So 'as?' in guard statement can not be done :)

Comment (zenangst) [reply Sources/Utils/Extensions/AVAsset+Extensions.swift]: Aha, that is interesting :)

**Comment by zenangst** [comment]
@geor-kasapidi thank you so much for the PR, this is great! cheers mate 🍺
